src/Microsoft.VisualStudio.AppDesigner/ApplicationDesigner/ApplicationDesignerWindowPane.vb

```diff
@@ -436,7 +436,8 @@ Namespace Microsoft.VisualStudio.Editors.ApplicationDesigner
                             hr = rdt.FindAndLockDocument(CUInt(_VSRDTFLAGS.RDT_NoLock), ProjectFullName, Hierarchy, ItemId, punkDocData, dwCookie)
                             If VSErrorHandler.Succeeded(hr) Then
                                 'We only want to save the project file itself, not any other its children (any of the other files in the project)
-                                hr = rdt.SaveDocuments(CUInt(__VSRDTSAVEOPTIONS.RDTSAVEOPT_SaveIfDirty Or __VSRDTSAVEOPTIONS.RDTSAVEOPT_SaveNoChildren),
+                                'but we do want to force it so that the MyApplication file will be saved is necessary
+                                hr = rdt.SaveDocuments(CUInt(__VSRDTSAVEOPTIONS.RDTSAVEOPT_ForceSave Or __VSRDTSAVEOPTIONS.RDTSAVEOPT_SaveNoChildren),
```

Comment:
This actually fixes a bug in the current system, when changing something on a property page that changes the .myapp file and not the project file, we mark the properties tab as dirty but when save is pressed the `SaveIfDirty` flag means it doesn't actually save, so the .myapp file doesn't save either. If the user then closes VS they get asked to save it, but from experience I can tell you that when VS asks you to save a file that you didn't manually change, you get used to clicking "Don't Save", and in this case you would lose changes. Forcing a save here means at least the running document table events will fire.